### PR TITLE
Adding the ability to detect and save PackageDependencies for a Run

### LINKED ISSUE #316: Record package dependencies when creating a run
**Is your feature request related to a problem? Please describe.**
When deploying an ML.NET model in a container, we need to know what package dependencies were used during training. E.g. was a `Microsoft.ML.FastTree` package dependencies added? If these are not included in the deployment environment then we will most likely have a run-time issue.

**Describe the solution you'd like**
Create a new data construct, `PackageDependency` with the following columns
- PackageDependencyId (Guid, PK)
- RunId (Guid, FK required)
- PackageName (string, required)
- PackageVersion (string, required)

Add a collection of package dependencies on a Run but no virtual navigation property to a Run from a PackageDependency (just the FK).

In the various `CreateRun` methods, discover all `Microsoft.ML` dependencies loaded into the app-domain and record them for a given run.

### PR DESCRIPTION
**Pull request by aslotte**
## Resolves
Fixes #316 

## Description
This PR adds the ability to automatically identify and log any `Microsoft.ML` dependencies during a run. This is important not only for auditability to know what version of ML.NET and what dependencies we used, but also as we can then install all of these dependencies automatically later when building out Docker container.

### CODE REVIEW

**Comment by aslotte** [comment]
@AnoojNair  / @Brett-Parker / @willvelida  - this PR is ready for a review. Thanks!

File: src/MLOps.NET.SQLServer/Migrations/20201001214039_AddPackageDependency.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace MLOps.NET.SQLServer.Migrations
+{
+    /// <summary>
+    ///
```
Comment (Brett-Parker) [inline src/MLOps.NET.SQLServer/Migrations/20201001214039_AddPackageDependency.cs:7]: Should these summaries be empty?

File: test/MLOps.NET.Tests/Catalogs/LifeCycleCatalogTests.cs [hunk 2]
```diff
@@ -91,7 +98,29 @@ public async Task CreateRunAsync_WithoutGitCommitHash_ShouldProvideEmptyGitCommi
...
+        [TestMethod]
+        public async Task CreateRunAsync_WithDependencies_ShouldProvidePackageDependencies()
+        {
+            //Arrange
+            var dependencies = new List<PackageDependency>
+            {
+                new PackageDependency
+                {
+                    Name = "Microsoft.ML",
+                    Version = "1.5.2"
```
Comment (Brett-Parker) [inline test/MLOps.NET.Tests/Catalogs/LifeCycleCatalogTests.cs:113]: Does this mean we will need to update the version every update?

**Review by Brett-Parker — APPROVED** [review]
Good work. Few minor comments. Can ignore

File: src/MLOps.NET.SQLServer/Migrations/20201001214039_AddPackageDependency.cs (hunk 1 above)
Comment (aslotte) [reply src/MLOps.NET.SQLServer/Migrations/20201001214039_AddPackageDependency.cs:7]: This particular ones are just for the migrations, and as the solution as a whole enforces comments on public classes they are required but not really needed...so left them empty :)

File: test/MLOps.NET.Tests/Catalogs/LifeCycleCatalogTests.cs (hunk 2 above)
Comment (aslotte) [reply test/MLOps.NET.Tests/Catalogs/LifeCycleCatalogTests.cs:113]: When we end up updating our ML.NET dependency yep. I couldn't find a better way to ensure that we actually read the proper dependencies than to use a real world use case for our tests. Hopefully it shouldn't be too much work, and be pretty obvious when it occurs.

**Comment by aslotte** [comment]
Thank you for the review @Brett-Parker!

**Pull request merged**